clientconn_test.go

```diff
@@ -820,3 +820,143 @@ func TestBackoffCancel(t *testing.T) {
 	cc.Close()
 	// Should not leak. May need -count 5000 to exercise.
 }
+
+// CONNECTING -> READY -> TRANSIENT FAILURE -> CONNECTING -> TRANSIENT FAILURE -> CONNECTING -> TRANSIENT FAILURE
+func TestDialCloseStateTransition(t *testing.T) {
+	defer leakcheck.Check(t)
+
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening. Err: %v", err)
+	}
+	defer lis.Close()
+	testFinished := make(chan struct{})
+	backoffCaseReady := make(chan struct{}, 1)
```

Comment:
Can this be an unbuffered struct that is closed instead?

Reply:
Done